lisa/transformers/kernel_installer.py

```diff
@@ -138,10 +138,14 @@ def _internal_run(self) -> Dict[str, Any]:
         self._log.info(f"installed kernel version: {installed_kernel_version}")
 
         # for ubuntu cvm kernel, there is no menuentry added into grub file
-        if hasattr(installer.runbook, "source"):
-            if installer.runbook.source != "linux-image-azure-fde":
-                posix = cast(Posix, node.os)
-                posix.replace_boot_kernel(installed_kernel_version)
+        if (
```

Comment:
hasattr is not a good check, use `isinstance` instead. Add comments in the code. Only if the comments applies to changes, it won't be put in the code.

Reply:
Sorry, Chi. I didn't get your point. This logic checks if it has "type" in the runbook. I'm not sure how `instance ` can achieve this function.

Reply:
For example, the `source` field is in RepoInstallerSchema, so it can be like `isinstance(installer, RepoInstaller)`. The `type` should be always here, and it doesn't need to be checked.

Reply:
Done